### PR TITLE
Update README with examples + edit text

### PR DESCRIPTION
**Pull request by mesameki**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -148,18 +149,22 @@ jupyter notebook
 --->
 
 # <a name="models"></a>
+# Supported Models
 
-# Models
+Any models that are trained on datasets in Python `numpy.array`, `pandas.DataFrame`, `iml.datatypes.DenseData`, or `scipy.sparse.csr_matrix` format are supported by this API.
```
Comment (wamartin-aml) [inline README.md]: It might be a bit confusing to say "Any models that are trained on datasets...are supported by this API" and then say in the next paragraph that we only work for models that provide `predict` or `predict_proba`. Maybe "This API supports models that are trained on datasets in Python [datatypes] format."?

File: README.md [hunk 2]
```diff
@@ -173,19 +178,264 @@ The explanation functions accept both models and pipelines as input. If a model
...
+    x_train, x_test, y_train, y_test = train_test_split(breast_cancer_data.data,            
+                                                        breast_cancer_data.target,  
+                                                        test_size=0.2,
+                                                        random_state=0)
+    clf = svm.SVC(gamma=0.001, C=100., probability=True)
+    model = clf.fit(x_train, y_train)
+    ```
+
+2. Call the explainer locally: To initialize an explainer object, you need to pass your model and some training data to the explainer's constructor. You can also optionally pass in feature names and output class names (if doing classification) which will be used to make your explanations and visualizations more informative. Here is how to instantiate an explainer object using `TabularExplainer`, `MimicExplainer`, and `PFIExplainer` locally. `TabularExplainer` is calling one of the four SHAP explainers underneath (`TreeExplainer`, `DeepExplainer`, `LinearExplainer`, or `KernelExplainer`), and is automatically selecting the most appropriate one for your use case. You can however, call each of its four underlying explainers directly.
```
Comment (wamartin-aml) [inline README.md]: Maybe `MimicExplainer`, **or** `PFIExplainer`

Comment (wamartin-aml) [inline README.md]: "is calling one of the four" -> "calls one of the four"

Comment (wamartin-aml) [inline README.md]: "`KernelExplainer`), and is automatically selecting" -> "`KernelExplainer`) and automatically selects"

File: README.md [hunk 3]
```diff
@@ -173,19 +178,264 @@ The explanation functions accept both models and pipelines as input. If a model
...
+# you can use the training data or the test data here
+global_explanation = explainer.explain_global(x_train)
+
+# if you used the PFIExplainer in the previous step, use the next line of code instead
+# global_explanation = explainer.explain_global(x_train, true_labels=y_test)
+
+# sorted feature importance values and feature names
+sorted_global_importance_values = global_explanation.get_ranked_global_values()
+sorted_global_importance_names = global_explanation.get_ranked_global_names()
+dict(zip(sorted_global_importance_names, sorted_global_importance_values))
```
Comment (wamartin-aml) [inline README.md]: I think this line is exactly the same as the call to `get_feature_importance_dict()` below

File: README.md [hunk 4]
```diff
@@ -173,19 +178,264 @@ The explanation functions accept both models and pipelines as input. If a model
...
+
+    ```python
+    from interpret.ext.blackbox import PFIExplainer 
+    
+    # "features" and "classes" fields are optional
+    explainer = PFIExplainer(model, 
+                             features=breast_cancer_data.feature_names, 
+                             classes=classes)
+    ```
+## Overal (Global) feature importance values
```
Comment (bethz) [inline README.md]: typo on overal

File: README.md [hunk 5]
```diff
@@ -173,19 +178,264 @@ The explanation functions accept both models and pipelines as input. If a model
...
 <a name=Example></a>
 
-# Example
+# Use Interpret-Community
+
+
+## Interpretability in training
+
+
+1. Train your model in a local Jupyter notebook.
```
Comment (bethz) [inline README.md]: Great job!

We've overloaded the word local and need to clarify so users don't get confused. In this content it's local compute, ie laptop.
Further down, we have a distinction for global vs local feature importance. We could say that local measures focus on the contribution of features for a specific prediction, whereas global measures take all predictions into account.
  
Perhaps change this instance to something like:
Train your model in a Jupyter notebook running on your local laptop.
And then note an alternative may be on a cloud.  
Then further down, we can highlight the difference between global and local feature importance.

File: README.md [hunk 6]
```diff
@@ -148,18 +149,22 @@ jupyter notebook
 --->
 
 # <a name="models"></a>
+# Supported Models
 
-# Models
+Any models that are trained on datasets in Python `numpy.array`, `pandas.DataFrame`, `iml.datatypes.DenseData`, or `scipy.sparse.csr_matrix` format are supported by this API.
+
+The explanation functions accept both models and pipelines as input. If a model is provided, the model must implement the prediction function predict or predict_proba that conforms to the Scikit convention.  If a pipeline script is provided, the explanation function assumes that the running pipeline script returns a prediction. The repository also supports models trained via PyTorch, TensorFlow, and Keras deep learning frameworks.
```
Comment (vingu) [inline README.md]: Rephrase this as
The explanation functions accept both models and pipelines as input, as long as the model or pipeline implements the prediction function predict or predict_proba that conforms to the Scikit convention.

Comment (vingu) [reply README.md]: On "The repository also supports models trained via PyTorch, TensorFlow, and Keras deep learning frameworks.", @imatiach-msft can you suggest edit on this? We support these because we provide wrapper on these model to make their API Scikit compliant?

File: README.md [hunk 7]
```diff
@@ -173,19 +178,264 @@ The explanation functions accept both models and pipelines as input. If a model
...
+sorted_local_importance_names = local_explanation.get_ranked_local_names()
+sorted_local_importance_values = local_explanation.get_ranked_local_values()
+```
+
+
+## Raw feature transformations
+
+Optionally, you can pass your feature transformation pipeline to the explainer to receive explanations in terms of the raw features before the transformation (rather than engineered features). If you skip this, the explainer provides explanations in terms of engineered features.
+
+The format of supported transformations is same as the one described in [sklearn-pandas](https://github.com/scikit-learn-contrib/sklearn-pandas). In general, any transformations are supported as long as they operate on a single column and are therefore clearly one to many.
```
Comment (vingu) [inline README.md]: I think we support many-to-many mapping too by turn on one setting. Forget about the parameter name. @hchandola

Comment (hchandola) [reply README.md]: we do. Its called `allow_all_transformations`

File: README.md (hunk 5 above)
Comment (mesameki) [reply README.md]: Very great points. Applied.

File: README.md (hunk 6 above)
Comment (mesameki) [reply README.md]: @vingu I believe this is true. As long as they can wrap it up for us to conform to predict and predict proba, we can support them. We did that for TFWorld.
